### PR TITLE
Skill +x tag and plural posting card bug fixes

### LINKED ISSUE #566: Skill "+x" tag should show all tags on hover instead of click
action: change to hover

We also want to add this popover behaviour in all the places where skills is displayed:

Account Edit/Create page

### PR DESCRIPTION
**Pull request by Boyazhang1**
## Ticket link

Closes #566 #571

## Implementation description
* admin posting card displays "1 volunteer" when singular instead of "volunteers"
* +x tag show all tags on hover instead of on click

Screen Shot 2022-08-13 at 5 02 06 PM

## Steps to test
1. Sign in as a volunteer user and navigate to http://localhost:3000/edit-account, select more than 3 skills and languages and verify the "+x" tag works on hover
Screen Shot 2022-08-15 at 3 49 35 PM

2. Sign in as an admin user, navigate to http://localhost:3000/admin/schedule/posting/1, select a time on the calendar with available volunteers and verify the skills +x tag works on hover
Screen Shot 2022-08-15 at 3 53 06 PM

3. Navigate to http://localhost:3000/admin/users and open the side panel for a volunteer user, verify the skills +x tag works on hover
Screen Shot 2022-08-15 at 3 55 03 PM

## What should reviewers focus on?
* the behavior of the +x tags work as intended, the number displayed in the +x tag is correct

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/yarn.lock [hunk 1]
```diff
@@ -2514,17 +2514,17 @@
...
+    "@tanstack/table-core" "8.5.11"
 
-"@tanstack/table-core@8.5.0":
-  version "8.5.0"
-  resolved "https://registry.yarnpkg.com/@tanstack/table-core/-/table-core-8.5.0.tgz#ccc52175ff4510ebe5ae0514131bdcd2d17b9ab2"
-  integrity sha512-YqyY28DyYQy5tmhmR9BYilVvszJn5ivHHxtpFIRrtxyoJIRLFske6jV8+Pz2zV8wkucfTBubqpp/IqhYWoJ4fA==
+"@tanstack/table-core@8.5.11":
+  version "8.5.11"
+  resolved "https://registry.yarnpkg.com/@tanstack/table-core/-/table-core-8.5.11.tgz#a23178a097df4b0b64bdfa6f79e6d8933e97c7f7"
+  integrity sha512-ZN61ockLaIAiiPbZfMKT2S03nbWx28OHg/nAiDnNfmN4QmAMcdwVajPn2QQwnNVGAr4jS4nbhbYzCcjq8livXQ==
```
Comment (neelismail01) [inline frontend/yarn.lock:2527]: Could you revert these changes before merging?

File: frontend/package.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "@fullcalendar/timegrid": "^5.10.1",
     "@rjsf/bootstrap-4": "^2.5.1",
     "@rjsf/core": "^2.5.1",
-    "@tanstack/react-table": "^8.5.0",
+    "@tanstack/react-table": "^8.5.11",
```
Comment (neelismail01) [inline frontend/package.json:21]: Was the change in package version necessary for this PR?

File: docker-compose.yml [hunk 3]
```diff
@@ -45,7 +46,7 @@ services:
     environment:
       - POSTGRES_MULTIPLE_DATABASES=${POSTGRES_DB_DEV},${POSTGRES_DB_TEST}
     healthcheck:
-      test: ["CMD-SHELL", "pg_isready -U postgres"]
+      test: ['CMD-SHELL', 'pg_isready -U postgres']
```
Comment (neelismail01) [inline docker-compose.yml:49]: Could you revert these changes to keep our Docker files consistent?

File: backend/Dockerfile [hunk 4]
```diff
@@ -11,4 +11,5 @@ RUN yarn install
 COPY . ./
 
 EXPOSE 5000
+EXPOSE 5555
```
Comment (neelismail01) [inline backend/Dockerfile:14]: Could you revert this file as well for consistency?

**Review by neelismail01 — CHANGES_REQUESTED** [review]

File: frontend/package.json (hunk 2 above)
Comment (Boyazhang1) [reply frontend/package.json:21]: oh I think that auto updated when I did yarn add lol, I'll manually revert it back

Comment (sherryhli) [reply frontend/package.json]: Could we revert the corresponding changes in yarn.lock as well?

File: frontend/src/components/admin/AdminPostingCard.tsx [hunk 5]
```diff
@@ -125,7 +125,7 @@ const AdminPostingCard = ({
               <Icon as={BsPeopleFill} w={6} pr={2} />
               {status === PostingFilterStatus.DRAFT
                 ? "Not accepting registrants"
-                : `${numVolunteers} volunteers`}
+                : `${numVolunteers} volunteer${numVolunteers > 1 ? "s" : ""}`}
```
Comment (MatoPlus) [inline frontend/src/components/admin/AdminPostingCard.tsx:128]:
```suggestion
                : `${numVolunteers} volunteer${numVolunteers != 1 ? "s" : ""}`}
```
edge case for num = 0 (0 volunteers)

File: frontend/src/components/common/TagsPopover.tsx [hunk 6]
```diff
@@ -12,18 +12,24 @@ import {
...
+  height?: string;
+  displayLength: number;
   tags: React.ReactElement[];
 };
 
 const TagsPopover: React.FC<TagsPopoverProps> = ({
   header,
   tags,
+  variant,
+  height,
```
Comment (MatoPlus) [inline frontend/src/components/common/TagsPopover.tsx:25]: lint

**Review by MatoPlus — CHANGES_REQUESTED** [review]
Looks good aside from some suggestions

**Review by MatoPlus — APPROVED** [review]
LGTM, thanks for getting this done!

**Pull request merged**